### ISSUE TITLE
How to force usage of libiconv during native compilation?

### ISSUE THREAD

**Issue by Ronson1909** [Entware-ng/Entware-ng#532]
I'm trying to compile some other project's sources (https://github.com/horchi/linux-p4d) on my QNAP ARMv5 device. The sources use iconv to convert character sets ISO8859-1 to UTF8 (include <iconv.h>). Compilation works, but during runtime it does not work (error EINVAL). It seems to use the libc version of iconv and I assume this causes a problem or is not supporting all necessary character sets?

For testing a minimum program that wants to use iconv is compiled using the g++ options mentioned in https://github.com/Entware-ng/Entware-ng/wiki/Using-gcc-(native-compilation) , i.e. when using

`-Wl,-rpath=/opt/lib -Wl,--dynamic-linker=/opt/lib/ld-linux.so.3 -L/opt/lib`

The ldd output is:

`ldd ./a.out
        libstdc++.so.6 => /opt/lib/libstdc++.so.6 (0xb6da3000)
        libm.so.6 => /opt/lib/libm.so.6 (0xb6cef000)
        libgcc_s.so.1 => /opt/lib/libgcc_s.so.1 (0xb6cd4000)
        libc.so.6 => /opt/lib/libc.so.6 (0xb6b90000)
        /opt/lib/ld-linux.so.3 (0xb6f0c000)`

So it's not using libiconv. However, I installed the libiconv(-full) package. Maybe it can be fixed by using libiconv!? How can I force usage of libiconv during the compilation of the program? The sources do not have a configure script. Do I need a different iconv.h than the one supplied with http://pkg.entware.net/binaries/armv5/include/include.tar.gz ? Or a compiler option?

Thanks

**Comment by zyxmon** [comment]
Native compilation is not supported. Sorry.
Use cross-compilation on a computer. In this case you must add libiconv include path and library to compilation flags.
It may work also in your case of native compilation.

**Comment by Ronson1909** [comment]
I found the solution. For whom it may concern:

It was necessary to:
1. add -liconv to the compiler arguments. that was quite clear and obvious and I did so, but the library was not used. This was due to 2.
2. and
  a) either replace the iconv.h of the downloaded (and extracted to /opt/include) include files (from http://pkg.entware.net/binaries/armv5/include/include.tar.gz) with the one from https://github.com/Entware-ng/Entware-ng/blob/master/package/libs/libiconv/src/include/iconv.h . This means that for other project compilations that require iconv point 1 needs to be considered (compiler flag)
  b) or put it in /opt/include with a different name (e.g. libiconv.h) and in your source include this file instead of iconv.h
